mempool/reactor.go

```diff
@@ -233,9 +226,12 @@ func (memR *Reactor) broadcastTxRoutine(peer p2p.Peer) {
 
 		// ensure peer hasn't already sent us this tx
 		if _, ok := memTx.senders.Load(peerID); !ok {
-			// send memTx
-			msg := &TxMessage{Tx: memTx.tx}
-			success := peer.Send(MempoolChannel, cdc.MustMarshalBinaryBare(msg))
+			msg := gogotypes.BytesValue{Value: []byte(memTx.tx)}
+			bz, err := msg.Marshal()
+			if err != nil {
+				panic(err)
```

Comment:
maybe just log the error? is this critical enough to crash Tendermint?

Reply:
to keep with the previous behavior a panic would be needed. if we arent able to marshal the type then there could be a larger issue at play. I can add a log here on top of the panic

Reply:
> I can add a log here on top of the panic

no need